libraries/Microsoft.Bot.Builder.Dialogs/DialogManager.cs

```diff
@@ -98,13 +94,9 @@ public async Task<DialogManagerResult> OnTurnAsync(ITurnContext context, Cancell
             await lastAccessProperty.SetAsync(context, lastAccess, cancellationToken: cancellationToken).ConfigureAwait(false);
 
             // get dialog stack 
+            var dialogsProperty = conversationState.CreateProperty<DialogState>(DIALOGS);
             DialogState dialogState = await dialogsProperty.GetAsync(context, () => new DialogState(), cancellationToken: cancellationToken).ConfigureAwait(false);
 
-            var namedScopes = MemoryScope.GetScopesMemory(context);
-            namedScopes[ScopePath.USER] = await userScope.GetAsync(context, () => new Dictionary<string, object>(StringComparer.InvariantCultureIgnoreCase), cancellationToken: cancellationToken).ConfigureAwait(false);
-            namedScopes[ScopePath.CONVERSATION] = await conversationScope.GetAsync(context, () => new Dictionary<string, object>(StringComparer.InvariantCultureIgnoreCase), cancellationToken: cancellationToken).ConfigureAwait(false);
-            namedScopes[ScopePath.TURN] = new Dictionary<string, object>(StringComparer.InvariantCultureIgnoreCase);
-
```

Comment:
this didn't make any sense. It added a hard dependency on how user and conversation state get initialized.  I moved this into BotStateMemoryScope and we are completely free to change the implementation of what resolves "user" => memory or "Conversation" => memory or any other scope for that matter

Reply:
Oh that makes sense